walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,79 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ArrayList;
+
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {  
+    long meeting_duration = request.getDuration();
+    Collection<String> meeting_attendees = request.getAttendees();
+
+    //First, find all incompatible time ranges so that we can use that as reference
+    //for finding time ranges which do actually work.
+    ArrayList<TimeRange> incompatible_ranges = findIncompatibleRanges(events, meeting_attendees);
+
+    //Sort the incompatible ranges by start time.
+    Collections.sort(incompatible_ranges, TimeRange.ORDER_BY_START);
+    Collection<TimeRange> available_ranges = new ArrayList<TimeRange>();
+
+    //Check if there is enough time between the end of the last event and the beginning of the next
+    //event. This is initalized to 0 since we are checking at the start of the day.
+    int meeting_request_start_time = 0; 
+    for (TimeRange event_range : incompatible_ranges){
+      int range_end = event_range.end();
```

Comment:
I would maybe change these variables to something more specific, like `incompatibleRangeStart` and `incompatibleRangeEnd` to make it more obvious how they're used. WDYT?